test/zzz-rsyslog.sh

```diff
@@ -320,9 +321,122 @@ else
 fi
 
 # switch LOGGING_FILE_PATH to console
-oc set env daemonset/rsyslog LOGGING_FILE_PATH=console
+oc set env $rsyslog_ds LOGGING_FILE_PATH=console
 os::cmd::try_until_failure "oc get pods $rpod > /dev/null 2>&1"
 os::cmd::try_until_success "oc get pods 2> /dev/null | grep -q 'rsyslog.*Running'"
 rpod=$( get_running_pod rsyslog )
 oc logs $rpod -c rsyslog > $ARTIFACT_DIR/rsyslog.console.log 2>&1
 os::cmd::expect_failure "grep 'oc exec <pod_name> -- logs' $ARTIFACT_DIR/rsyslog.console.log"
+
+# switch back LOGGING_FILE_PATH to default
+oc set env $rsyslog_ds LOGGING_FILE_PATH-
+os::cmd::try_until_failure "oc get pods $rpod > /dev/null 2>&1"
+os::cmd::try_until_success "oc get pods 2> /dev/null | grep -q 'rsyslog.*Running'"
+rpod=$( get_running_pod rsyslog )
+
+artifact_log "************************"
+artifact_log "Testing undefined fields"
+artifact_log "************************"
+
+artifact_log "0. Preparing test data"
+cfg=$( mktemp )
+cat > $cfg << EOF
+# DEBUGGING
+set $!undefined1 = "undefined1";
+set $!undefined11 = "1111";
+set $!undefined12 = "True";
+set $!empty1 = "";
+set $!undefined2!undefined2 = "undefined2";
+set $!undefined1!undefined22 = "2222";
+set $!undefined2!undefined23 = "False";
+set $!undefined3!emptyvalue = "";
+set $!undefined4 = "undefined4";
+set $!undefined5 = "undefined5";
+set $!undefined.6 = "undefined5";
+# DEBUGGING
+EOF
+# add our test data to the rsyslog pipeline via a volume mount
+oc set volumes $rsyslog_ds --add --name=rsyslog-undefined-test \
+    -t hostPath -m /etc/rsyslog.d/66-debug.conf \
+    --path $cfg 2>&1 | artifact_out
+
+restart_rsyslog_pod $rpod
+rpod=$( get_running_pod rsyslog )
+
+artifact_log "1. default values"
+os::cmd::expect_success_and_text "oc oc exec $rpod -c rsyslog -- grep "use_undefined:" /var/log/rsyslog/rsyslog.log | tail -n 1 | | awk '{print $3}'" "false"
```

Comment:
no `$3` here?

Reply:
hmm - looks like some other copy/paste errors.

Reply:
```
os::cmd::expect_success_and_text "oc exec $rpod -c rsyslog -- grep 'use_undefined:' /var/log/rsyslog/rsyslog.log | tail -n 1 | | awk '{print $3}'" "false"
```

Reply:
Thanks.  Yeah, the lines were broken... :(
For some reason, my attempt to run the CI from get-cluster-run-tests fail in the earlier section.  So, I decided to debug here.

```
177 sleep 10
178 wait_for_fluentd_to_catch_up get_logmessage get_logmessage2 
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```